libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentType.java

```diff
@@ -151,7 +151,8 @@ public static XContentType fromMediaTypeOrFormat(String mediaType) {
     private static String removeVersionInMediaType(String mediaType) {
         if (mediaType.contains("vnd.elasticsearch")) {
             return mediaType.replaceAll("vnd.elasticsearch\\+", "")
-                .replaceAll("\\s*;\\s*compatible-with=\\d+", "");
+                .replaceAll("\\s*;\\s*compatible-with=\\d+", "")
+                .replaceAll("\\s*;\\s*",";"); // to allow matching using startsWith
```

Comment:
not sure I understand the comment ?

Reply:
matching in fromMediaTypeOrFormat can use startsWith so that 
`application/json; charset=utf-8` can be matched. however this assumes that `;` is immediately following `mediaTypeWithoutParameters`

I don't want to refactor this in 7.x, the intention of ` .replaceAll("\\s*;\\s*",";");` was to make it slightly easier if you provided `application/vnd.elasticsearch+json;compatible-with=7 ; charset=utf-8` with accidental space.

Reply:
I have updated the comment with a reference to a method